### DISCUSSION TITLE
[OpenSource] Authentification with OAuth. How to create an admin user.

### DISCUSSION THREAD

**Discussion by malte-baumann** [Unleash/unleash#7979] [Q&A] [answered]
I am implementing an auth-hook in my case with keycloak. Similar to [your Examples](https://github.com/Unleash/unleash-examples/blob/main/v4/securing-keycloak-auth/keycloak-auth-hook.js). 
That does work, I can successfully login with SSO. 
My Question: 
Is it possible to create an admin user this way? How can I elevate my role to admin?

**Comment by gastonfournier** [comment, marked as answer]
Hi @malte-baumann you can use the user service to create an admin email, something like:
```typescript
await userService.createUser({
                'Your Name',
                'youremail@yourdomain.com',
                RoleName.ADMIN,
            });
```

The code is available here: https://github.com/Unleash/unleash/blob/064744b18f6608095cc2e151956d466e4e3cd4ba/src/lib/services/user-service.ts#L212-L215

You see that you can also send in the password, but having it in code might be a security risk, but there are ways.

Let me know if that helps! Also, remember that you can use our community slack: https://unleash-community.slack.com/
